datameta/security/__init__.py

```diff
@@ -309,3 +303,27 @@ def revalidate_admin(request):
     if user.site_admin or user.group_admin:
         return user
     raise HTTPUnauthorized()
+
+
+def successfully_authenticated(**kwargs) -> None:
+    """Function to invoke side effects after successful authentications."""
+
+    def clear_login_attemtps(kwargs):
+        """Sets number of login attempts to 0."""
+
+        def get_failed_logins_within_hour(db) -> int:
```

Comment:
The nesting is somewhat confusing and doesn't really do anything here since the reset is anyway the only thing the function does. Maybe remove the functions for now and only encapsulate if we ever need to add more actions after successful auths?

Reply:
I agree, I'm also not happy with this